frontend/src/old-pages/Clusters/JobSubmitDialog.tsx

```diff
@@ -145,55 +146,137 @@ function JobCostEstimate() {
+    !detail || detail === '' ? clearState([...jobPath, property]) : setState([...jobPath, property], detail);
+  }
+
+  return <>
+    <SpaceBetween direction="vertical" size="xxs" key={property}>
+      <Header 
+        variant="h2" 
+        description={description}>
+        {header}
+      </Header>
+      <FormField>
+        <Input 
+          onChange={({ detail }) => {setStateIfNotEmpty(detail.value);}}
+          value={useState([...jobPath, property])}
+          placeholder={placeholder}
+          disabled={disabled}
+        />
+      </FormField>
+    </SpaceBetween>
+  </>
+}
+
 export default function JobSubmitDialog({
   submitCallback
 }: any) {
   const { t } = useTranslation();
   const open = useState([...submitPath, 'dialog']);
   const error = useState([...submitPath, 'error']);
-  const jobPath = [...submitPath, 'job'];
   const clusterName = getState(['app', 'clusters', 'selected']);
+  const clusterPath = ['clusters', 'index', clusterName];
 
-  const job = useState(jobPath);
+  const job: {[key: string]: string} = useState(jobPath);
   const submitting = useState([...submitPath, 'pending']);
-  let jobName = useState([...jobPath, 'job-name']);
-  let chdir = useState([...jobPath, 'chdir']);
-  let nodes = useState([...jobPath, 'nodes']);
-  let ntasks = useState([...jobPath, 'ntasks']);
-  let mem = useState([...jobPath, 'mem']);
-  let command = useState([...jobPath, 'command']);
-  let wrap = useState([...jobPath, 'wrap']) || false;
+  let jobType: string = useState([...submitPath, 'job-entry']);
+
+  React.useEffect(() => jobTypeSelect('command'), []);
 
   let isMemBasedSchedulingEnabled = useState(
-      ['clusters', 'index', clusterName, 'config', 'Scheduling', 'SlurmSettings', 'EnableMemoryBasedScheduling']
+    [...clusterPath, 'config', 'Scheduling', 'SlurmSettings', 'EnableMemoryBasedScheduling']
   ) || false;
 
+  let isSlurmApiEnabled = getScripts(
+    useState([...clusterPath, 'config', 'HeadNode', 'CustomActions'])
+  ).includes('slurm-rest-api');
+
+  const isScriptSelected = useState([...submitPath, 'script']);
+
+  const jobTypeSelect = (entryType: string) => {
+    setState([...submitPath, 'job-entry'], entryType);
+    entryType === 'command' ? setState([...jobPath, 'wrap'], true) : clearState([...jobPath, 'wrap']);
+    if(entryType === 'script') {
+      setState([...submitPath, 'script'], true);
+      clearState([...jobPath, 'command']);
+      clearState([...jobPath, 'mem']); // TODO: Memory based scheduling not currently supported in backend for Slurm API
+    } else {
+      clearState([...submitPath, 'script']);
+    }
+  }
+
+  const scriptSelectDescription = isSlurmApiEnabled ? "" : "Requires Slurm REST API";
+
+  const jobTypeHeader = {
+    'command': "Command",
+    'file': "Script Path"
+  }[jobType]
+
+  const jobTypeDescription = {
+    'command': "The command to run as a part of this job.",
+    'file': "Path to the script to run."
+  }[jobType]
```

Comment:
issue(blocking): Careful, by hardcoding strings this way we would not be able to support different languages in the app. Please use translation tokens instead, you can see [here an example](https://github.com/aws-samples/pcluster-manager/pull/175/commits/fdc6b77987c87a26f51dbc8da5d371d95ef80601)